device-registry/src/main/scala/org/genivi/sota/device_registry/db/DeviceRepository.scala

```diff
@@ -105,6 +106,15 @@ object DeviceRepository {
       _ <- devices.insertOrUpdate(newDevice)
     } yield DeviceSeen(device.namespace, uuid, now)
 
+  // inner Future can fail with java.util.NoSuchElementException
+  def setActivated(uuid: Uuid)
+                  (implicit ec: ExecutionContext): DBIO[Device] =
+    for {
+      unactivatedDevice <- findByUuid(uuid).filter(_.activatedAt.isEmpty)
+      activatedDevice = unactivatedDevice.copy(activatedAt = Some(Instant.now()))
+      _ <- devices.insertOrUpdate(activatedDevice)
+    } yield activatedDevice
```

Comment:
You could also run the update, if you have a query to find by uuid AND activatedAt.isEmpty and the `.map` to `activatedAt` and `.update`. Then you could use `handleSingleUpdateError` to check only one row was updated.

If you chose to use the find/update, you could also use `insertOrUpdateWithKey`.